src/Workspace/ErrorWorkspacePlaceholder.jsx

```diff
@@ -1,21 +1,33 @@
 import React from 'react';
+import { useHistory } from 'react-router-dom/cjs/react-router-dom';
+import isEnabled from '../helpers/featureFlags';
 import './Workspace.less';
 import NotFoundSVG from '../img/not-found.svg';
 import { components } from '../params';
+import workspaceRegistrationConfig from '../localconf';
 
-class ErrorWorkspacePlaceholder extends React.Component {
-  render() {
-    const supportEmail = components.login?.email || 'support@datacommons.io';
-    return (
-      <div className='error-workspace-placeholder__error-msg'>
-        <h1>Error opening workspace...</h1>
-        <p>
-        Workspace access requires authorization. Please contact <a href={`mailto:${supportEmail}`}>{supportEmail}</a> for more information.
-        </p>
-        <NotFoundSVG />
-      </div>
-    );
+const ErrorWorkspacePlaceholder = () => {
+  const history = useHistory();
+  if (isEnabled('workspaceRegistration')) {
+    if (workspaceRegistrationConfig) {
+      // eslint-disable-next-line react/prop-types
```

Comment:
Why disable this?

Reply:
Hey Mingfei, after you mentioned this, and the using of useEffect() , I felt like we can use of Redirect here again instead of history.push, because, when I tried to do useEffect, I see the entire component first and then being redirected to the form. I will make a new push and let you know